packages/flutter/test/services/fake_image_provider.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui show Codec;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+/// An image provider implementation for testing that is using a [ui.Codec]
+/// that it was given at construction time (typically the job of real image
+/// providers is to resolve some data and instantiate a [ui.Codec] from it).
+class FakeImageProvider extends ImageProvider<FakeImageProvider> {
```

Comment:
Random thought / feel free to ignore: can this just be an anonymous mockito?

Reply:
This is more a fake than a mock, we have a bunch of ImageProvider subclasses that instantiates the image stream in different ways. This fake is another ImageProvider subclass that is just handed a ready codec at construction time.

To replace this with a mock we'll have to build a partial mock (as we want the code from the base ImageProvider unmocked). IMO that will feel a bit dirty.